src/components/encoding-pane/property-editor-schema.ts

```diff
@@ -181,9 +279,59 @@ export function generateFormData(shelfId: ShelfId, fieldDef: ShelfFieldDef) {
       const prop = customProp.prop;
       const nestedProp = customProp.nestedProp;
       const propertyKey = nestedProp ? prop + '_' + nestedProp : prop;
-      formData[propertyKey] = fieldDef[prop] ? nestedProp ? fieldDef[prop][nestedProp] : fieldDef[prop] : undefined;
+      const fData = fieldDef[prop] ? nestedProp ? fieldDef[prop][nestedProp] : fieldDef[prop] : undefined;
+      // handle empty input for text input
+      formData[propertyKey] = fData === '?' ? '' : fData;
     }
   }
 
   return formData;
 }
+
+function filterEnumElements(fieldDef: ShelfFieldDef, prop: string, nestedProp: string, possibleVals: string[]) {
```

Comment:
This method is very convoluted.

Reply:
I figured having a generic helper method filter enum values based on the fieldDef type would be helpful 

The intended purpose of this method is to filter enum values based on the `fieldDef`'s type. So for instance, only appropriate scale types are displayed, or appropriate color schemes based on the definition of the `fieldDef`.

Reply:
> I could do this in generatePropertyEditorSchema method as well if having a separate method isn't needed.

Yep, that's exactly what you should do.  In general you want to minimize code jump when reading the code, unless you have some common methods that are repetitive.

In this case, they are not truly repetitive, but you made them the same function.  So this basically add unnecessary complexity in two ways: (1)  unnecessary code jump and (2)more conditional logic to distinguish things that are not supposed to be in the same method.

I could agree that another good reason for a code jump is when a method is too long, but I still disagree with this way of method extraction.